cirrus/server/cirrus/main.py

```diff
@@ -109,6 +110,41 @@ def initialize_glean():
     return pings, metrics
 
 
+def get_metrics_data(
+    enrolled_partial_configuration: Dict[str, Any]
+) -> List[tuple[str, str, str, str]]:
+    events: List[Dict[str, Any]] = enrolled_partial_configuration.get("events", [])
+    data: List[tuple[str, str, str, str]] = []
+    for event in events:
+        if event.get("change") == "Enrollment":
+            experiment_slug: str = event.get("experiment_slug", "")
+            branch_slug: str = event.get("branch_slug", "")
+            enrollment_id: str = event.get("enrollment_id", "")
+            experiment_type: str = app.state.remote_setting.get_recipe_type(
+                experiment_slug
+            )
+            data.append((experiment_slug, branch_slug, enrollment_id, experiment_type))
+    return data
+
+
+def record_metrics(enrolled_partial_configuration: Dict[str, Any], client_id: str):
+    metrics = get_metrics_data(
+        enrolled_partial_configuration=enrolled_partial_configuration
+    )
+    for experiment_slug, branch_slug, enrollment_id, experiment_type in metrics:
+        app.state.metrics.cirrus_events.enrollment.record(
+            app.state.metrics.cirrus_events.EnrollmentExtra(
+                user_id=client_id,
+                app_id=app_id,
+                experiment=experiment_slug,
+                branch=branch_slug,
+                enrollment_id=enrollment_id,
+                experiment_type=experiment_type,
+            )
+        )
+    app.state.pings.enrollment.submit()
```

Comment:
Are there any internal Glean pings (baseline, metrics, etc) that Cirrus relies on? If so, it might be useful to call the Glean experiment annotation API, which can mark other pings, including internal ones, with the active experiments. If the only pings Cirrus relies on are the custom pings defined here, then this may not be necessary.

More information about the Glean Experiment API can be found here: https://mozilla.github.io/glean/book/reference/general/experiments-api.html

Reply:
I think we rely only on the custom pings so we _should_ be okay. Good callout though!